### PR TITLE
Enable lost debug-only code

### PR DESCRIPTION
**Pull request by rainersigwald**
At some point we stopped defining _DEBUG for debug builds, so all this conditional code was never being built.

```sh-session
$ rep -r "#if _DEBUG" "#if DEBUG" *.cs
1993 files processed. 9 files changed. 14 replacements.
```

### CODE REVIEW

**Comment by rainersigwald** [comment]
This won't pass until #4893 is merged.

File: src/Tasks/UnregisterAssembly.cs [hunk 1]
```diff
@@ -4,6 +4,7 @@
 #if FEATURE_APPDOMAIN
 
 using System;
+using System.Diagnostics;
```
Comment (BenVillalobos) [inline src/Tasks/UnregisterAssembly.cs]: Is the using necessary for this PR?

Comment (rainersigwald) [reply src/Tasks/UnregisterAssembly.cs]: It is: the calls to `[System.Diagnostics.]Debug.Assert` in the `#if`s I changed wouldn't compile without it.

**Pull request merged**